src/lints/repr_packed_added.ron

```diff
@@ -0,0 +1,72 @@
+SemverQuery(
+    id: "repr_packed_added",
+    human_readable_name: "repr(packed) added",
+    description: "A struct or union has been marked with #[repr(packed)].",
+    required_update: Major,
+    reference_link: Some("https://doc.rust-lang.org/cargo/reference/semver.html#repr-packed-add"),
+    query: r#"
+    {
+        CrateDiff {
+            baseline {
+                item {
+                    ... on ImplOwner {
+                        type: __typename @filter(op: "one_of", value: ["$types"]) @output @tag
+                        visibility_limit @filter(op: "=", value: ["$public"])
+
+                        attribute @fold @transform(op: "count") @filter(op: "=", value: ["$zero"]) {
+                            content {
+                                base @filter(op: "=", value: ["$repr"])
+                                argument {
+                                    base @filter(op: "=", value: ["$packed"])
+                                }
+                            }
+                        }
+
+                        importable_path {
+                            path @tag @output
+                            public_api @filter(op: "=", value: ["$true"])
+                        }
+                    }
+                }
+            }
+            current {
+                item {
+                    ... on ImplOwner {
+                        __typename @filter(op: "=", value: ["%type"])
+                        visibility_limit @filter(op: "=", value: ["$public"])
+                        name @output
+
+                        attribute {
+                            content {
+                                base @filter(op: "=", value: ["$repr"])
+                                argument {
+                                    base @filter(op: "=", value: ["$packed"])
+                                }
+                            }
+                        }
+
+                        importable_path {
+                            path @filter(op: "=", value: ["%path"])
+                            public_api @filter(op: "=", value: ["$true"])
+                        }
+
+                        span_: span @optional {
+                            filename @output
+                            begin_line @output
+                        }
+                    }
+                }
+            }
```

Comment:
Minor perf nit here that you couldn't possibly have known about, because I need to write more docs 😅

Trustfall has no prior knowledge of how often `repr(packed)` structs are, so it will run the `baseline` and `current` legs of this query in lexical order ("in the order they appear in the query"). But the query will run faster if we do the more-rare case first (since it will prune more aggressively and do less work). We as Rustaceans know that most structs aren't `repr(packed)`. The `current` block looks for `repr(packed)` structs, so doing it first will make the query run faster.

Reply:
In principle we could have added this knowledge ("`repr(packed)` structs are rare, prioritize that") into the adapter, but we haven't yet and it's likely not worth the effort since this isn't something that comes up often.